test/python/apps/exchange.py

```diff
@@ -79,11 +96,37 @@ def __init__(self,
         self._partner_name: bytes = b"Default_Partner"
         self._transaction: bytes = b""
         self._payout_payload: bytes = b""
-        self._refund_payload: bytes = b""
+        self._refund_payload: Optional[bytes] = None
+
         if self._subcommand == SubCommand.SWAP:
             self._curve = ec.SECP256K1()
-        else:
+            self._signature_computation = Signature_Computation.BINARY_ENCODED_PAYLOAD
```

Comment:
Looking at the size and the repetition, it may be the time to create a struct just to delegate the subcommand specifications, something like:

```python
@dataclass(frozen=True)
class SubCommandSpecs:
    curve: ec._EllipticCurve
    signature_computation: SignatureComputation
    payload_encoding: PayloadEncoding
    transactionType: Union[NewTransactionResponse, NewSellResponse, NewFundResponse]
    required_fields: List[str]
    payout_field: str
    refund_field: Optional[str]
```

Reply:
But then we would have to explicit every field anyway when declaring the class instance right ?

Reply:
At initialization? I don't think so. Keyword arguments are still ordered and can be filled without being named. It's only mandatory when changing the order or eluding arguments.
```python
> SubCommandSpecs(1, 2, 3, 4, 5, 6, 7)
  SubCommandSpecs(curve=1, signature_computation=2, payload_encoding=3, transactionType=4, required_fields=5, payout_field=6, refund_field=7)
```